lib/chef/knife/azure_base.rb

```diff
@@ -125,6 +125,75 @@ def pretty_key(key)
 
       # validate command pre-requisites (cli options)
       def validate_params!
+        if locate_config_value(:winrm_password) && (locate_config_value(:winrm_password).length <= 6 && locate_config_value(:winrm_password).length >= 72)
+          ui.error("The supplied password must be 6-72 characters long and meet password complexity requirements")
+          exit 1
+        end
+
+        if locate_config_value(:ssh_password) && (locate_config_value(:ssh_password).length <= 6 && locate_config_value(:ssh_password).length >= 72)
```

Comment:
Because of the `<=` and `>=` this would prevent passwords that are 6 or 72 characters long even though the error says the password could be 6 or 72 characters long.